app/assets/javascripts/discourse/app/components/tag-info.js

```diff
@@ -4,39 +4,43 @@ import { and, reads } from "@ember/object/computed";
 import { service } from "@ember/service";
 import { htmlSafe } from "@ember/template";
 import { isEmpty } from "@ember/utils";
+import { tagName } from "@ember-decorators/component";
 import { ajax } from "discourse/lib/ajax";
 import { popupAjaxError } from "discourse/lib/ajax-error";
 import discourseComputed from "discourse-common/utils/decorators";
 import I18n from "discourse-i18n";
 
-export default Component.extend({
-  dialog: service(),
-  tagName: "",
-  loading: false,
-  tagInfo: null,
-  newSynonyms: null,
-  showEditControls: false,
-  canAdminTag: reads("currentUser.staff"),
-  editSynonymsMode: and("canAdminTag", "showEditControls"),
-  editing: false,
-  newTagName: null,
-  newTagDescription: null,
-  router: service(),
+@tagName("")
+export default class TagInfo extends Component {
+  @service dialog;
+  @service router;
+
+  loading = false;
+  tagInfo = null;
+  newSynonyms = null;
+  showEditControls = false;
+
+  @reads("currentUser.staff") canAdminTag;
+  @and("canAdminTag", "showEditControls") editSynonymsMode;
+
+  editing = false;
+  newTagName = null;
+  newTagDescription = null;
```

Comment:
just a grouping tweak

```suggestion
  editing = false;
  newTagName = null;
  newTagDescription = null;

  @reads("currentUser.staff") canAdminTag;
  @and("canAdminTag", "showEditControls") editSynonymsMode;
```